### PR TITLE
[Dashboard] Logical view backend for dashboard

### PR DESCRIPTION
**Pull request by zzyunzhi**
## Why are these changes needed?

Display dashboard with actor-tree structure. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/dashboard/dashboard.py [hunk 1]
```diff
@@ -281,6 +289,32 @@ def get_node_stats(self) -> Dict:
...
+                    addr = (worker_info["ipAddress"], worker_info["port"])
+                    if addr in self._addr_to_actor_id:
+                        print("worker is not Driver", addr)
+                        actor_id = self._addr_to_actor_id[addr]
+                        flattened_tree[actor_id].update(worker_info)
+
+        actor_tree = flattened_tree
+        for actor_id, parent_id in child_to_parent.items():
+            actor_tree[parent_id]["children"][actor_id] = actor_tree[actor_id]
+        return actor_tree["root"]["children"]
```
Comment (ericl) [inline python/ray/dashboard/dashboard.py]: Would be good to think about how to write a unit test for this rendering function.

File: src/ray/core_worker/transport/direct_actor_transport.cc [hunk 2]
```diff
@@ -310,4 +310,13 @@ void CoreWorkerDirectTaskReceiver::HandleDirectActorCallArgWaitComplete(
...
 
+int64_t CoreWorkerDirectTaskReceiver::SizeSchedulingQueue() {
+  int64_t size_scheduling_queue = 0;
+  for (auto const& it : scheduling_queue_) {
+    size_scheduling_queue += it.second->NumPendingTasks();
+  }
+  RAY_LOG(INFO) << "YYY Size of Scheduling Queue" << size_scheduling_queue;
+  return size_scheduling_queue; 
+}
+
```
Comment (ericl) [inline src/ray/core_worker/transport/direct_actor_transport.cc]: Consider adding a test to test_direct_actor_transport.cc that exercises the new stats code.

File: python/ray/dashboard/dashboard.py [hunk 3]
```diff
@@ -159,6 +158,9 @@ def get_forbidden(_) -> aiohttp.web.Response:
 
         async def raylet_info(req) -> aiohttp.web.Response:
             D = self.raylet_stats.get_raylet_stats()
+            print("XXX")
```
Comment (pcmoritz) [inline python/ray/dashboard/dashboard.py]: This should be removed

File: python/ray/dashboard/dashboard.py [hunk 4]
```diff
@@ -171,6 +173,9 @@ def get_forbidden(_) -> aiohttp.web.Response:
                     extra_info.append("{}: {} / {}".format(
                         resource_name, occupied, total))
                 data["extraInfo"] = ", ".join(extra_info)
+                # test logical view
+                print(actor_tree)
```
Comment (pcmoritz) [inline python/ray/dashboard/dashboard.py]: The print should be removed

File: python/ray/dashboard/dashboard.py (hunk 1 above)
Comment (pcmoritz) [reply python/ray/dashboard/dashboard.py]: Yeah, we can do this in a test:
```
info = ray.init(include_webui=True)
response = requests.get(info["webui_url"] + "/api/raylet_info")
# Check response.content
```

File: src/ray/core_worker/transport/direct_actor_transport.h [hunk 5]
```diff
@@ -324,6 +324,10 @@ class SchedulingQueue {
     ScheduleRequests();
   }
 
+  int64_t NumPendingTasks() {
```
Comment (pcmoritz) [inline src/ray/core_worker/transport/direct_actor_transport.h]: This should be removed

File: src/ray/protobuf/core_worker.proto [hunk 6]
```diff
@@ -125,6 +125,13 @@ message GetCoreWorkerStatsRequest {
 message GetCoreWorkerStatsReply {
   // String displayed on Web UI.
   string webui_display = 1;
+  // Length of the task queue. 
+  int64 task_queue_length = 2;
```
Comment (pcmoritz) [inline src/ray/protobuf/core_worker.proto]: Can you add comments for all these fields?

File: src/ray/core_worker/core_worker.cc [hunk 7]
```diff
@@ -1036,6 +1040,19 @@ void CoreWorker::HandleGetCoreWorkerStats(const rpc::GetCoreWorkerStatsRequest &
                                           rpc::GetCoreWorkerStatsReply *reply,
                                           rpc::SendReplyCallback send_reply_callback) {
   reply->set_webui_display(webui_display_);
+  reply->set_task_queue_length(num_tasks_accepted_ - num_tasks_executed_);
+  reply->set_current_executed_task("test");
```
Comment (pcmoritz) [inline src/ray/core_worker/core_worker.cc]: This needs to be fixed before we can merge the PR.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**